src/main/java/com/allocadia/carbonite/CarbonitePersist.java

```diff
@@ -0,0 +1,111 @@
+package com.allocadia.carbonite;
+
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.support.JdbcDaoSupport;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+
+import com.allocadia.carbonite.utils.QueryUtils;
+import com.google.common.base.Joiner;
+
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.SneakyThrows;
+
+public class CarbonitePersist<T extends PersistedObject> extends JdbcDaoSupport {
+
+    private final PersistedObjectCache poc;
+    private final Carbonite carbonite;
+    
+    public CarbonitePersist(PersistedObjectCache poc, Carbonite carbonite) {
+        this.poc = poc;
+        this.carbonite = carbonite;
+    }
+    
+    public Collection<T> save(Collection<T> objects) {
+        ModifiedObjects modifiedObjects = ModifiedObjects.from(objects);
+        Class<?> clazz = objects.iterator().next().getClass();
+        PersistenceInfo<PersistedObject> info = (PersistenceInfo<PersistedObject>) carbonite.getPersistenceInfo(clazz);
+        
+        persistNew(modifiedObjects.getNewObjects(), info);
+        persistDirty(modifiedObjects.getDirtyObjects(), info);
+    }
+
+    @SneakyThrows
+    private void persistDirty(List<PersistedObject> dirtyObjects, PersistenceInfo<PersistedObject> info) {
+        for (int i = 0; i < dirtyObjects.size(); i++) {
+            PersistedObject dirtyObject = dirtyObjects.get(i);
+            List<String> dirtyFields = new ArrayList<>(dirtyObject.getDirty());
```

Comment:
It doesn't help that this method is incomplete, but have a look at persistNew()'s prepared statement bit.  We need to reference the position of the field - it isn't a case of iteration order